easybuild/easyconfigs/c/ctffind5/ctffind5-5.0.2-foss-2023a.eb

```diff
@@ -0,0 +1,81 @@
+# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2024/05
+easyblock = 'ConfigureMake'
+
+name = 'ctffind5'
+version = '5.0.2'
+
+local_commit = 'b21db55a91366fe4f75301c56091d213bbd326eb'
+local_branch = 'ctffind5_merge'
+
+homepage = 'https://grigoriefflab.umassmed.edu/ctf_estimation_ctffind_ctftilt'
+description = """Program for finding CTFs of electron micrographs."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+# Use git_config in order to allow setting CISTEM_CURRENT_BRANCH and CISTEM_VERSION_TEXT by configure
+sources = [{
+    'download_filename': '%s.tar.gz' % local_commit,
+    'filename': 'cisTEM-ctffind5-%s.tar.gz' % local_commit[:7],
+    'git_config': {
+        'url': 'https://github.com/timothygrant80',
+        'repo_name': 'cisTEM',
+        'commit': local_commit,
+        'keep_git_dir': True,
+    }
+}]
```

Comment:
Hi @ThomasHoffmann77
this recipe is a piece of work. Congratulation - it was not an easy one. 

About the source, can it work with an approach similar to the following?
```
source_urls = ['https://github.com/lh3/bwa/archive']
sources = [{'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ}]
checksums = ['be460d6e13ddf34896aafae00bad71e05a0b9f7e23490eeeca8ad257065f5e60']
```

In this way we can include the checksum in the recipe and help with the reproducibility.

Reply:
@kErica the configure script makes use of "git branch" and similar in order to inject the branch name  and version and therefore requires the .git directory from the git_config checkout.